test/integration/tests/upload/files/FakeHackage.hs

```diff
@@ -1,8 +1,32 @@
 {-# LANGUAGE OverloadedStrings #-}
 
-import Network.Simple.TCP
+import Control.Concurrent
+import Control.Monad
+import Network.Socket hiding (recv)
+import Network.Socket.ByteString (recv, sendAll)
+import System.Exit
+import System.IO
 
 -- | Fake server that always responds with HTTP OK
 main =
-    serve (Host "127.0.0.1") "12415" $ \(socket, _) ->
-        send socket "HTTP/1.1 200 OK\r\n\r\n"
+    withSocketsDo $ do
+        _ <- forkIO serve
+        -- Exit after a delay to ensure the process doesn't linger around
+        threadDelay 10000000
+        exitSuccess
+
+serve :: IO ()
+serve = do
+    let hints = defaultHints {addrFlags = [AI_PASSIVE], addrSocketType = Stream}
+    (addr:_) <- getAddrInfo Nothing Nothing (Just "12415")
```

Comment:
`hints` seems used nowhere, what's it doing there? [Docs](http://hackage.haskell.org/package/network-2.7.0.2/docs/Network-Socket.html#v:getAddrInfo) suggest calling `getAddrInfo (Just hints) ...` to actually use it. That affects #4191!